### PR TITLE
Inclue wallet state manager

### PR DESCRIPTION
**Pull request by tomossomot**
And all Wallet functionality accessible via CLI

### CODE REVIEW

**Comment by lawlawlaw** [comment]
test_make_server fails

**Comment by tomossomot** [comment]
Update:
- Catch panic from invalid input
- Generate fresh Seed when not in testing mode
- Allow setting of Daemon file system address from config

**Comment by lawlawlaw** [comment]
Result of cargo test:

failures:

---- daemon::tests::test_make_server stdout ----
thread 'daemon::tests::test_make_server' panicked at 'assertion failed: format!("{}", serde_json :: to_string(& request) .
        unwrap()).contains("Connection refused")', client/src/daemon.rs:244:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    daemon::tests::test_make_server

**Comment by tomossomot** [comment]
> Result of cargo test:
> 
> failures:
> 
> ---- daemon::tests::test_make_server stdout ----
> thread 'daemon::tests::test_make_server' panicked at 'assertion failed: format!("{}", serde_json :: to_string(& request) .
> unwrap()).contains("Connection refused")', client/src/daemon.rs:244:9
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> 
> failures:
> daemon::tests::test_make_server

This test passes for me. Are you running a wallet daemon process somewhere?

**Comment by lawlawlaw** [comment]
> > Result of cargo test:
> > failures:
> > ---- daemon::tests::test_make_server stdout ----
> > thread 'daemon::tests::test_make_server' panicked at 'assertion failed: format!("{}", serde_json :: to_string(& request) .
> > unwrap()).contains("Connection refused")', client/src/daemon.rs:244:9
> > note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> > failures:
> > daemon::tests::test_make_server
> 
> This test passes for me. Are you running a wallet daemon process somewhere?

no

**Comment by lawlawlaw** [comment]
> > Result of cargo test:
> > failures:
> > ---- daemon::tests::test_make_server stdout ----
> > thread 'daemon::tests::test_make_server' panicked at 'assertion failed: format!("{}", serde_json :: to_string(& request) .
> > unwrap()).contains("Connection refused")', client/src/daemon.rs:244:9
> > note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> > failures:
> > daemon::tests::test_make_server
> 
> This test passes for me. Are you running a wallet daemon process somewhere?

The error is: 

{"Err":{"Generic":"IoError(Os { code: 2, kind: NotFound, message: \"No such file or directory\" })"}}'

Is there a file missing?

**Comment by lawlawlaw** [comment]
> > Result of cargo test:
> > failures:
> > ---- daemon::tests::test_make_server stdout ----
> > thread 'daemon::tests::test_make_server' panicked at 'assertion failed: format!("{}", serde_json :: to_string(& request) .
> > unwrap()).contains("Connection refused")', client/src/daemon.rs:244:9
> > note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> > failures:
> > daemon::tests::test_make_server
> 
> This test passes for me. Are you running a wallet daemon process somewhere?

It fails in make_unix_conn_call, here: 

let client = UnixConnection::<JsonCodec<DaemonRequest, DaemonResponse>>::new(&daemon_address, JsonCodec::new()).wait()?;

daemon_address is /tmp/rustd.sock which does not exist on my system

I suppose the file will be created when the daemon is run, and therefore the above error should be permitted in the test assertion.

**Comment by lawlawlaw** [comment]
There are errors compiling integration-tests

**Comment by lawlawlaw** [comment]
Some integration tests are failing

**Comment by tomossomot** [comment]
> Some integration tests are failing

Which ones? I think they work for me.

**Comment by lawlawlaw** [comment]
This one:
test batch_transfer_test::tests::test_batch_sigs ... thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("error sending request for url (http://localhost:8000/ecdsa/keygen/first): operation timed out")', integration-tests/src/lib.rs:185:63
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FAILED

File: server/src/protocol/conductor.rs [hunk 1]
```diff
@@ -1078,7 +1078,7 @@ mod tests {
         let mut db = MockDatabase::new();
         db.expect_set_connection_from_config().returning(|_| Ok(()));
 
-        let mut seq = Sequence::new();
+        // let seq = Sequence::new();
```
Comment (lawlawlaw) [inline server/src/protocol/conductor.rs:1081]: Remove commented code

**Review by lawlawlaw — APPROVED** [review]

File: server/src/protocol/conductor.rs (hunk 1 above)
Comment (tomossomot) [reply server/src/protocol/conductor.rs]: I wasnt sure if you wanted to keep it for future.

Comment (tomossomot) [reply server/src/protocol/conductor.rs]: removed

**Pull request merged**